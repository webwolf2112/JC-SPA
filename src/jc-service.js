const axios = require('axios');

const API_PATH = `http://localhost:8005/api/systemusers`;

export const getUsers = async() => {
    try {
      const response = await axios.get(API_PATH);
      return response.data.results;

    } catch (error) {
        console.error( error );
      return { error };
    }
}

export const createUser = async (formValues ) => {

    try {
        return await axios.post(API_PATH, formValues);
    } catch (error) {
        console.error( error );
        return { error };
    }

}

export const updateUser = async(id, formValues) => {
    try {
      return await axios.put(`${API_PATH}/${id}`, formValues);
    } catch (error) {
        console.error( error );
        return { error };
    }
}

export const getUserById = async(id) => {
    try {
      const response = await axios.get(`${API_PATH}/${id}`);
      return response.data.results;

    } catch (error) {
        console.error( error );
        return { error };
    }
}

export const deleteUser = async(id) => {
    try {
      return await axios.delete(`${API_PATH}/${id}`);
    } catch (error) {
      return { error };
    }
}
