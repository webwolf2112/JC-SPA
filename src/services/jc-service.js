const axios = require('axios');

const API_PATH = `http://localhost:8005/api/systemusers`;

export const getUsers = async() => {
    try {
      const response = await axios.get(API_PATH);
      return response.data.results;

    } catch (error) {
      return { error };
    }
}

export const createUser = async (formValues ) => {

    try {
        return await axios.post(API_PATH, formValues);
    } catch (error) {
        console.error( error );
        return { error };
    }

}

export const updateUser = async(id, formValues) => {
    try {
      return await axios.put(`${API_PATH}/${id}`, formValues);
    } catch (error) {
        console.error( error );
        return { error };
    }
}

/**This was not used. If I were to use it I would have added in an additional modal that shows all 
 * of the user information in the UI.
**/
export const getUserById = async(id) => {
    try {
      const response = await axios.get(`${API_PATH}/${id}`);
      return response.data.results;

    } catch (error) {
        return { error };
    }
}

export const deleteUser = async(id) => {
    try {
      return await axios.delete(`${API_PATH}/${id}`);
    } catch (error) {
      return { error };
    }
}
