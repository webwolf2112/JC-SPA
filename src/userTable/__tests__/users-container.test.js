import React from 'react';
import { shallow, mount } from 'enzyme';
import UsersContainer from '../users-container';
import { getUsers } from '../../services/jc-service';
import { act } from 'react-dom/test-utils'

jest.mock( '../../services/jc-service');

describe( 'Users Container', () => {

    beforeEach( () => {
        jest.clearAllMocks();
    });

    it('Should render correct', () => {
        const wrapper = shallow(
          <UsersContainer />,
        );
      
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render correct', async () => {
        getUsers.mockReturnValue( Promise.resolve( [] ));
            let wrapper; 
            await act(async () => {
                wrapper = mount(
                    <UsersContainer />,
                    );
            });
            expect( getUsers ).toHaveBeenCalled();
    });


    /** These are the additional test cases I would normally write. 
        1) test each method, update, delete, and create for the error message to show
        2) test each method, update, delete, and create called the correct endpoints
        3) test that the loading screen shows before the initial getUsers call
      **/
});