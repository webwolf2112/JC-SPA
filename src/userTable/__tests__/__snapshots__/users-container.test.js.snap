// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Container Should render correct 1`] = `
<div
  className="loading"
>
   Loading . . . 
</div>
`;
